contracts/simple_dex/lib.rs

```diff
@@ -399,6 +412,8 @@ mod simple_dex {
 
             balance_token_out
                 .checked_sub(op4)
+                // If the division is not even, leave the 1 unit of dust in the exchange instead of paying it out.
+                .and_then(|result| result.checked_sub(if op3 % op2 > 0 { 1 } else { 0 }))
```

Comment:
Why are we always subtracting 1, even if the remainder of `op3 / op2` divitions (`op3 % op2`) may be higher than that.

Reply:
+1, maybe indeed subtract the remainder

Reply:
So my thinking here was:

1. The "real" value of `op3 / op3` is between `op4` and `op4 + 1`.
2. The whole number division always gives us `op4`.
3. So to be more conservative, but as close to the "real" value as possible we use `op4 + 1` when the division is not even.

I don't think `op3 % op2` factors into this at all. Am I completely missing something?